### PR TITLE
Encrypt cloud credentials

### PR DESCRIPTION
**Pull request by themylogin**
Ticket: #29020

### CODE REVIEW

File: gui/freeadmin/models/fields.py [hunk 1]
```diff
@@ -57,6 +59,28 @@ def to_python(self, value):
...
+        return notifier().pwenc_encrypt(json.dumps(value))
+
+    def from_db_value(self, value, expression, connection, context):
+        return self.to_python(value)
+
+    def to_python(self, value):
+        if not value:
+            return {}
+        if isinstance(value, str):
+            return json.loads(notifier().pwenc_decrypt(value))
```
Comment (william-gr) [inline gui/freeadmin/models/fields.py]: @themylogin Just to be sure, have you tested what happens if pwenc_decrypt fails? If that wont break backup.credential.query altogether?

**Comment by william-gr** [comment]
rebuild

**Comment by william-gr** [comment]
rebuild

**Comment by themylogin** [comment]
rebuild

**Comment by william-gr** [comment]
rebuild

**Comment by william-gr** [comment]
rebuild

**Comment by william-gr** [comment]
rebuild

**Comment by william-gr** [comment]
rebuild

**Pull request merged**